test/milton/MiltonEvents.t.sol

```diff
@@ -0,0 +1,525 @@
+        IpToken ipTokenDai = getIpTokenDai(address(daiMockedToken));
+        MockCase0Stanley stanleyDai = getMockCase0Stanley(address(daiMockedToken));
+        MiltonStorage miltonStorageDai = getMiltonStorage();
+        MockCase0MiltonDai miltonDai = getMockCase0MiltonDai(
+            address(daiMockedToken),
+            address(iporOracle),
+            address(miltonStorageDai),
+            address(_miltonSpreadModel),
+            address(stanleyDai)
+        );
+        MockCase0JosephDai josephDai = getMockCase0JosephDai(
+            address(daiMockedToken),
+            address(ipTokenDai),
+            address(miltonDai),
+            address(miltonStorageDai),
+            address(stanleyDai)
+        );
+        prepareApproveForUsersDai(_users, daiMockedToken, address(josephDai), address(miltonDai));
+        prepareMilton(miltonDai, address(josephDai), address(stanleyDai));
+        prepareJoseph(josephDai);
+        prepareIpToken(ipTokenDai, address(josephDai));
+        // when
+        _miltonSpreadModel.setCalculateQuotePayFixed(TestConstants.PERCENTAGE_6_18DEC); // 6%
+        vm.prank(_userOne);
+        iporOracle.itfUpdateIndex(address(daiMockedToken), TestConstants.PERCENTAGE_5_18DEC, block.timestamp); // 5%, PERCENTAGE_5_18DEC
+        vm.prank(_liquidityProvider);
+        josephDai.itfProvideLiquidity(TestConstants.USD_28_000_18DEC, block.timestamp); // TestConstants.USD_28_000_18DEC
+        vm.prank(_userTwo);
+        miltonDai.itfOpenSwapPayFixed(
+            block.timestamp, // openTimestamp
+            TestConstants.USD_10_000_18DEC, // totalAmount
+            TestConstants.PERCENTAGE_6_18DEC, // acceptableFixedInterestRate, 6%
+            TestConstants.LEVERAGE_18DEC // leverage
+        );
+        vm.prank(_userOne);
+        iporOracle.itfUpdateIndex(address(daiMockedToken), TestConstants.PERCENTAGE_160_18DEC, block.timestamp); // PERCENTAGE_160_18DEC
+        vm.prank(_userTwo);
+        vm.expectEmit(true, true, true, true);
+        emit CloseSwap(
+            1, // swapId
+            address(daiMockedToken), // asset
+            block.timestamp + TestConstants.SWAP_DEFAULT_PERIOD_IN_SECONDS, // closeTimestamp, 28 days, PERIOD_28_DAYS_IN_SECONDS
+            _userTwo, // liquidator
+            18957318804358692392282, // transferredToBuyer
+            TestConstants.ZERO, // transferredToLiquidator
+            TestConstants.TC_INCOME_TAX_18DEC // incomeFeeValue
+        );
+        miltonDai.itfCloseSwapPayFixed(
+            1, // swapId
+            block.timestamp + TestConstants.SWAP_DEFAULT_PERIOD_IN_SECONDS // closeTimestamp, 28 days, PERIOD_28_DAYS_IN_SECONDS
+        );
+    }
+
+    function testShouldEmitEventWhenClosePayFixedSwap6DecimalsAndTakerClosedSwap() public {
+        // given
+        MockTestnetToken usdtMockedToken = getTokenUsdt();
+        ItfIporOracle iporOracle = getIporOracleAsset(_userOne, address(usdtMockedToken), 0);
```

Comment:
Why you changed EMA to 0 ?? In old test EMA = 3%, please use TestConstants.TC_DEFAULT_EMA_18DEC ! Don't change in new test given params from old test, because you are changing assumption and test will potentially test totaly different behaviour.